### PR TITLE
Add Search Functionality

### PR DESCRIPTION
**Pull request by n00b4ever**
image
image

### CODE REVIEW

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java [hunk 1]
```diff
@@ -130,7 +132,29 @@ protected void onBindViewHolder(@NonNull BookListViewHolder holder, int position
...
-                source.putString(context.getString(R.string.book_click_source_fragment), context.getString(R.string.search_title));
+
+                /* If book belongs to owner, OR not a search match while searching, hide it */
+                if (book.getOwner().equals(UserCredentialAPI.getInstance().getUsername())
+                    || (!this.search.equals("") && !searchMatch(book)))
+                {
+                    holder.itemView.setVisibility(View.GONE);
+                    holder.itemView.setLayoutParams(new RecyclerView.LayoutParams(0, 0));
+                }
+
```
Comment (JawadHossain) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java:143]: can u pls remove the blank line add a comment on why the margin is set here?

**Review by KateVlaar — APPROVED** [review]

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/SearchFragment.java [hunk 2]
```diff
@@ -107,6 +115,81 @@ public void onStop()
...
+                if (searchBooksRecyclerAdapter.getSearch().length() != 0)
+                {
+                    searchBooksRecyclerAdapter.setSearch("");
+                    searchBooksRecyclerAdapter.updateOptions(options);
+                }
+                searchView.setQuery("", false);
+
+                // https://stackoverflow.com/questions/1109022/how-do-you-close-hide-the-android-soft-keyboard-using-java
+                InputMethodManager imm = (InputMethodManager) view.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+                imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
```
Comment (JawadHossain) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/SearchFragment.java:189]: do searchView.clearFocus(); instead

**Review by JawadHossain — CHANGES_REQUESTED** [review]
Please see the two minor change requests.

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/SearchFragment.java (hunk 2 above)
Comment (n00b4ever) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/SearchFragment.java]: oh, lol

**Review by JawadHossain — APPROVED** [review]

**Pull request merged**